dali/kernels/common/convert.h

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_COMMON_CONVERT_H_
+#define DALI_KERNELS_COMMON_CONVERT_H_
+
+#include <cuda_runtime.h>
+#include <limits>
+#include <type_traits>
+
+namespace dali {
+namespace kernels {
+
+template <typename T>
+struct const_limits;
+
+// std::numeric_limits are not compatible with CUDA
+template <typename T>
+__host__ __device__ constexpr T max_value() { return const_limits<T>::max; }
+template <typename T>
+__host__ __device__ constexpr T min_value() { return const_limits<T>::min; }
+
+#define DEFINE_TYPE_RANGE(type, min_, max_) template <>\
+struct const_limits<type> { static constexpr type min = min_, max = max_; };
```

Comment:
I'd remove the semicolon from the macro and end every instantiation with ';' down below. Otherwise some editors might go crazy when trying to auto indent